csrank/fate_ranking.py

```diff
@@ -593,55 +537,15 @@ def predict_scores(self, X, **kwargs):
             scores = self._predict_scores_fixed(X, **kwargs)
         return scores
 
-    def set_tunable_parameters(self, point):
-        FATERankingCore.set_tunable_parameters(self, point)
-        hidden_layers_created = False
-        self.logger_gorc = logging.getLogger(GENERAL_OBJECT_RANKING_CORE)
-        keys = self._tunable.copy().keys()
-        named = dict(zip(keys, point))
-
-        SET_LAYER_KEYS = [N_HIDDEN_SET_UNITS, N_HIDDEN_SET_LAYERS]
-        for key in keys:
-            if key not in SET_LAYER_KEYS + [REGULARIZATION_FACTOR]:
-                del named[key]
-
-        for name, param in named.items():
-            if name in SET_LAYER_KEYS + [REGULARIZATION_FACTOR]:
-                selected_params = {k: v for k, v in named.items()
-                                   if k in SET_LAYER_KEYS}
-                self.__dict__.update(selected_params)
-                self.kernel_regularizer = l2(l=named[REGULARIZATION_FACTOR])
-                if not hidden_layers_created:
-                    self._create_set_layers(
-                        activation=self.activation,
-                        kernel_initializer=self.kernel_initializer,
-                        kernel_regularizer=self.kernel_regularizer)
-                hidden_layers_created = True
-            else:
-                self.logger_gorc.warning(
-                    "This ranking algorithm does not support a tunable parameter called {}".format(name))
-
-    @classmethod
-    def tunable_parameters(cls):
-        logger = logging.getLogger(GENERAL_OBJECT_RANKING_CORE)
-        if cls._tunable is None:
-            FATERankingCore.tunable_parameters()
-        logger.info("Use Early Stopping {}".format(cls._use_early_stopping))
-        if cls._use_early_stopping:
-            cls._tunable[EARLY_STOPPING_PATIENCE] = EARLY_STOPPING_PATIENCE_DEFAULT_RANGE
-        dict_params = dict([
-            (N_HIDDEN_SET_UNITS, SET_HIDDEN_UNITS_DEFAULT_RANGE),
-            (N_HIDDEN_SET_LAYERS, SET_LAYERS_DEFAULT_RANGES),
-            (REGULARIZATION_FACTOR, REGULARIZATION_FACTOR_DEFAULT_RANGE)])
-        logger.info("Set params {}".format(print_dictionary(dict_params)))
-        for k, v in dict_params.items():
-            cls._tunable[k] = v
-        cls._tunable = OrderedDict(sorted(cls._tunable.items()))
-
-    @classmethod
-    def set_tunable_parameter_ranges(cls, param_ranges_dict):
-        logger = logging.getLogger(GENERAL_OBJECT_RANKING_CORE)
-        return tunable_parameters_ranges(cls, logger, param_ranges_dict)
+    def set_tunable_parameters(self, n_hidden_set_units=32, n_hidden_set_layers=2, **point):
+        FATERankingCore.set_tunable_parameters(self, **point)
```

Comment:
We should also offer the parameters of the superclass here.
Otherwise the users do not know that there are more parameters like `n_hidden_joint_layers`.

Reply:
done